doc/guides/device_mgmt/index.rst

```diff
@@ -39,14 +39,417 @@ functionality over BLE and serial.
+   and allow the new running image to self-confirm after any checks by calling
+   :c:func:`boot_write_img_confirmed`.
+
+.. tip::
+
+    The maximum size of a chunk communicated between the client and server is set
+    with :option:`CONFIG_IMG_MGMT_UL_CHUNK_SIZE`. The default is 512 but can be
+    decreased for systems with low amount of RAM downto 128. When this value is
+    changed, the ``mtu`` of the port must be smaller than or equal to this value.
+
+.. tip::
+
+    Building with :option:`CONFIG_IMG_MGMT_VERBOSE_ERR` enables better error
+    messages when failures happen (but increases the application size).
+
+.. _stats_mgmt:
+
+Statistics Management
+*********************
+
+Statistics are used for troubleshooting, maintenance, and usage monitoring; it
+consists basically of user-defined counters which are tightly connected to
+``mcumgr`` and can be used to track any information for easy retrieval.
+
+Statistics are organized in sections (also called groups), and each section can
+be individually queried. Defining new statistics sections is done by using macros
+available under ``<stats/stats.h>``. Each section consists of multiple variables
+(or counters), all with the same size (16, 32 or 64 bits).
+
+To create a new section ``my_stats``::
+
+  STATS_SECT_START(my_stats)
+    STATS_SECT_ENTRY(my_stat_counter1)
+    STATS_SECT_ENTRY(my_stat_counter2)
+    STATS_SECT_ENTRY(my_stat_counter3)
+  STATS_SECT_END
+
+  STATS_SECT_DECL(my_stats) my_stats;
+
+Each entry can be declared with ``STATS_SECT_ENTRY`` (or the equivalent
+``STATS_SECT_ENTRY32``), ``STATS_SECT_ENTRY16`` or ``STATS_SECT_ENTRY64``.
+All statistics in a section must be declared with the same size.
+
+The statistics counters can either have names or not, depending on the setting
+of the :option:`CONFIG_STATS_NAMES` option. Using names requires an extra
+declaration step (before ``STATS_SECT_DECL``)::
+
+  STATS_NAME_START(my_stats)
+    STATS_NAME(my_stats, my_stat_counter1)
+    STATS_NAME(my_stats, my_stat_counter2)
+    STATS_NAME(my_stats, my_stat_counter3)
+  STATS_NAME_END(my_stats)
+
+Disabling :option:`CONFIG_STATS_NAMES` will free resources.
+
+The final steps to use a statistics section is to initialize and register it::
+
+  rc = STATS_INIT_AND_REG(my_stats, STATS_SIZE_32, "my_stats");
+  assert (rc == 0);
+
+In the running code a statistics counter can be incremented by 1 using
+``STATS_INC``, by N using ``STATS_INCN`` or reset with ``STATS_CLEAR``.
+
+The sub-commands available are:
+
+``mcumgr <connection-options> stat list``: Read the list of stats sections from
+a device.
+
+``mcumgr <connection-options> stat <section-name>``: Display data for the given
+stats section.
+
+.. tip::
+
+   :option:`CONFIG_STAT_MGMT_MAX_NAME_LEN` sets the maximum length of a section
+   name that can can be accepted as parameter for showing the section data, and
+   might require tweaking for long section names.
+
+.. _fs_mgmt:
+
```

Comment:
I think that this section requires a little reorder: previous sections describe how to use mcumgr commands and what are effects of invocation; this section starts with code definition and then presents command.
I think that this should be reversed: first should be description of `mcumgr stat` command, followed by subsection that describes how to define such stats in code.

I am still not sure whether the howto on stats in code should be here, or should it get dedicated section, but for now I think that reversal of the help would be enough.

Reply:
Yeah, I reworked the section, there were also some other issues, which would result in build errors, and some things were not very clear, and put it all in the suggested format, should be much better now.